tests/validation/tests/v3_api/test_kdm_changes.py

```diff
@@ -51,36 +52,46 @@ def test_clusters_for_kdm():
     else:
         default_k8s_versions = [K8S_VERSION]
     list_process = []
-    plugin = PLUGIN
+    network_plugins = NETWORK_PLUGINS
+    dns_providers = DNS_PROVIDERS
     print("default_k8s_versions: ", default_k8s_versions)
     for k8s_version in default_k8s_versions:
         rke_config_new = deepcopy(rke_config)
         rke_config_new["kubernetesVersion"] = k8s_version
-        node_count = NODE_COUNT_KDM_CLUSTER * len(plugin)
+        node_count = NODE_COUNT_KDM_CLUSTER * len(network_plugins) * len(dns_providers) * 2
         aws_nodes = \
             AmazonWebServices().create_multiple_nodes(
                 node_count, random_test_name(HOST_NAME))
         i = 0
-        for plug in plugin:
-            if plug == "calico" or plug == "canal" or plug == "weave":
-                rke_config_new["network"]["options"] = \
-                    {"flannel_backend_type": "vxlan"}
-            rke_config_new["network"] = {"type": "networkConfig",
-                                         "plugin": plug}
-            client = get_user_client()
-            cluster_name = random_test_name(plug)
-            cluster = client.create_cluster(name=cluster_name,
-                                            driver="rancherKubernetesEngine",
-                                            rancherKubernetesEngineConfig=
-                                            rke_config_new)
-            p1 = multiprocessing.Process(target=validate_custom_cluster_kdm,
-                                         args=(cluster,
-                                               aws_nodes
-                                               [i:i+NODE_COUNT_KDM_CLUSTER]))
-            CLUSTER_LIST.append(cluster)
-            list_process.append(p1)
-            p1.start()
-            i = i + NODE_COUNT_KDM_CLUSTER
+        for network_plugin in network_plugins:
+            for dns_provider in dns_providers:
+                for nodelocaldns in True, False:
+                    if network_plugin == "calico" or network_plugin == "canal" or network_plugin == "weave":
+                        rke_config_new["network"]["options"] = \
+                            {"flannel_backend_type": "vxlan"}
+                    rke_config_new["network"] = {"type": "networkConfig",
+                                                 "plugin": network_plugin}
+                    rke_config_new["dns"] = {"type": "dnsConfig",
+                                                 "provider": dns_provider}
+                    cluster_options = network_plugin + "-" + dns_provider
+                    if nodelocaldns:
+                        rke_config_new["dns"]["nodelocal"] = \
+                            {"type": "nodelocal", "ipAddress": "169.254.20.10"}
+                        cluster_options += "-nodelocaldns"
```

Comment:
@superseb Why do we need `cluster_options += "-nodelocaldns"` here. ?

Reply:
I pass `cluster_options` to `cluster_name = random_test_name(cluster_options)` to make sure all the cluster options are in the name, which is then used to retrieve the enabled options in the formatted output below.